tex/generic/pgf/utilities/pgfutil-common-expl.tex

```diff
@@ -0,0 +1,80 @@
+% Copyright 2022 by Yukai Chou
+%
+% This file may be distributed and/or modified
+%
+% 1. under the LaTeX Project Public License and/or
+% 2. under the GNU Public License.
+%
+% See the file doc/generic/pgf/licenses/LICENSE for more details.
+
+%
+% This file provides utility commands that are drop-in replacements
+% for expl3 macros.
+%
+% Naming conventions:
+%   pgf_cs = "pgfexpl@" + expl3_cs.replace("_", "@").replace(":", "@@")
```

Comment:
I think with a little bit of work we could make `_` a letter. There are very few uses of `_` outside of `\write` and comments.

Reply:
`:` is a bit more difficult, because it's used in `\pgfutil@ifnextchar` (and some other stuff that needs to skip spaces).

Reply:
doable. Do you expect it happens in current PR?

Reply:
I'd say yes, because we should address this immediately as `pgfexpl` is introduced. However, if you can't do it in a timely manner, we can go ahead without it.

Reply:
What do you think about
```
% Naming conventions:
%   pgf_cs = "__pgfexpl_" + expl3_cs.replace(":", "@")
% For example,
%   expl3_cs: \exp_args:Ne,          \tl_put_right:Nn
%   pgf_cs:   \__pgfexpl_exp_args@Ne, \__pgfexpl_tl_put_right@Nn
```

Reply:
> `:` is a bit more difficult, because it's used in `\pgfutil@ifnextchar` (and some other stuff that needs to skip spaces).

Since the ultimate goal is to depend on `expl3-code.tex`, now I'm kind of leaning more towards making both `_` and `:` letters at once. A constant token list `\c__pgfexpl_semicolon_tl` and some careful expansion works will overcome the `:` issue.